src/media/hwschematic.ts

```diff
@@ -146,43 +164,175 @@ const findWidgetFormState = new FindWidgetFormData(
 
 initializeFindWidget(document, onAddNode, onAddPath, onClearSelection, findWidgetFormState);
 
+function getCurrentlySelectedIds(highlightGroups: any): Set<string> {
+	const currentlySelectedIds: Set<string> = new Set();
 
-/**
- * Render the document in the webview.
- */
-function updateContent(text: string) {
-	let json;
-	try {
-		if (!text) {
-			text = '{}';
+	for (const highlightGroup of highlightGroups) {
+		if (highlightGroup.isSelected) {
+			for (const id of highlightGroup.members) {
+			currentlySelectedIds.add(id);
+			}
 		}
-		json = JSON.parse(text);
-	} catch {
-		hwSchematic.setErrorText('Error: Document is not valid json');
-		return;
 	}
+
+	return currentlySelectedIds;
+}
+
+function addHighlightGroups(nodes: any, highlightGroups: any) {
+	findWidgetFormState.highlightGroups = [];
+	//findWidgetFormState.addToHighlightGroups("GROUP");
+
+	for (const highlightGroup of highlightGroups) {
+		const newHighlightGroup = new HighlightGroup(highlightGroup.name, findWidgetFormState);
+		newHighlightGroup.CheckboxChecked = highlightGroup.isSelected;
+		const membersSet = new Set(highlightGroup.members);
+
+		for (const node of nodes) {
+			if (membersSet.has(node.id)){
+				newHighlightGroup.addItem(node);
+			}
+		}
+		findWidgetFormState.highlightGroups.push(newHighlightGroup);
+	}
+}
+
+function updateContent(json: any, eventType: string) {
+	// funkcia ktora prepise groupy v json do jeho properties do root hwMeta //selection groups
+	// zo selection groups v json zostavime selection groups, zavolame apply highlight? a vykresli sa
+	// do selection groups v jason dame komponenty z objektu, ktory mame
+
+	// vysypeme currently selectcted, zo selection groups v documente pridame currently seleceted,
```

Comment:
Use English please, it is complicated to find people working on HW related project even on whole word. Just SK would be quite a constraint.